tests/validation/tests/v3_api/common.py

```diff
@@ -639,24 +686,59 @@ def validate_dns_record(pod, record, expected):
 
 
 def validate_dns_entry(pod, host, expected):
+    print("Expected target:", expected)
+
     # requires pod with `dig` available - TEST_IMAGE
     cmd = 'ping -c 1 -W 1 {0}'.format(host)
-    ping_output = kubectl_pod_exec(pod, cmd)
+    if is_windows_os:
```

Comment:
Same as `validate_http_response` comment above. We can determine the `is_windows_os` by the pod scheduling rule.

Reply:
Yes, create a function ` match_os(pod.scheduling, "windows")` to determine this.